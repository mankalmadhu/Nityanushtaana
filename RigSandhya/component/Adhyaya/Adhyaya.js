import React from 'react';
import { TouchableHighlight,View, Image, Text } from 'react-native';
import {Link } from 'react-router-native';

import adhyayaVinyasa from './Adhyaya.vinyasa';

export default class Adhyaya extends React.Component {
  constructor(props) {
    super(props);
  }
  
  render() {
   return (
      <Link to="/shubhamastu" component={TouchableHighlight}>
        <View style={adhyayaVinyasa.avarana}>
         <Image source={this.props.chitra} style={adhyayaVinyasa.chitra} />
         <Text style={adhyayaVinyasa.pathya}>
          {`${this.props.shershike}`}
         </Text>
         </View>
       </Link>
     );
  }
}