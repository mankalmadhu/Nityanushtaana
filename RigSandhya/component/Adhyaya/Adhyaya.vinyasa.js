import { StyleSheet } from 'react-native';

export default adhyayaVinyasa = StyleSheet.create({
  avarana: {
    flex: 1,
    padding: 16,
    flexDirection: 'row',
    alignItems: 'center',
  },
  pathya: {
    marginLeft: 12,
    fontSize: 16,
    fontFamily: 'ganapati',
    color: '#ffffff'
  },
  chitra: {
    height: 40,
    width: 40,
    borderRadius: 20,
  },
});