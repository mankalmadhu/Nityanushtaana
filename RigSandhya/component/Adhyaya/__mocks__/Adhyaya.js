import React from 'react';

export default Adhyaya = (props) => (
    <div>
      Adhyaya {`${props.shershike} ${props.chitra}`}
    </div>
);