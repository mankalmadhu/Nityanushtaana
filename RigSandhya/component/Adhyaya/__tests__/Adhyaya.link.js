import 'react-native';
import React from 'react';
import Renderer from 'react-test-renderer';

import Adhyaya from '../Adhyaya';

jest.mock('react-router-native', (props)=>{
  const MockReact = require('React');
  class Link extends MockReact.Component
  {
    render() {
      return MockReact.createElement('Link' , this.props.to);
    }
  }
  return ({Link : Link});
}) ;

const testData = {"shershike":"last"};
test("Adhyaya Link Test" , () => {
   const adhyaya = Renderer.create(
        <Adhyaya {...testData}/>
    ).toJSON();

   expect(adhyaya).toMatchSnapshot();
});    