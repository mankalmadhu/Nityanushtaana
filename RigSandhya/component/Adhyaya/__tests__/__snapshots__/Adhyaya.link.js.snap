// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adhyaya Link Test 1`] = `
<Link
  0="/"
  1="l"
  2="a"
  3="s"
  4="t"
/>
`;
